src/ucp/wireup/select.c

```diff
@@ -1875,31 +1893,37 @@ ucp_wireup_add_rma_bw_lanes(const ucp_wireup_select_params_t *select_params,
         bw_info.criteria.local_md_flags |= UCT_MD_FLAG_INVALIDATE_RMA;
     }
 
-    /* Find an AM lane that needs to be segregated from RMA accesses */
-    ucs_carray_for_each(lane_desc, select_ctx->lane_descs,
-                        select_ctx->num_lanes) {
-        if (!(lane_desc->lane_types & UCS_BIT(UCP_LANE_TYPE_AM))) {
-            continue;
-        }
+    /* Find if an AM lane needs to be separated from RMA operations. More lanes
+     * can cause fence calls to last longer at scale. Applications making heavy
+     * use of fence operations usually don't request the TAG feature, hence the
+     * check.
+     */
+    if (context->config.features & (UCP_FEATURE_TAG | UCP_FEATURE_AM)) {
+        ucs_carray_for_each(lane_desc, select_ctx->lane_descs,
+                            select_ctx->num_lanes) {
+            if (!(lane_desc->lane_types & UCS_BIT(UCP_LANE_TYPE_AM))) {
+                continue;
+            }
 
-        wiface = ucp_worker_iface(ep->worker, lane_desc->rsc_index);
-        flags  = ucp_wireup_iface_get_perf_attr_flags(wiface);
-        if (!(flags & UCT_PERF_ATTR_FLAGS_TX_RX_SHARED)) {
-            continue;
-        }
+            wiface = ucp_worker_iface(ep->worker, lane_desc->rsc_index);
+            flags  = ucp_wireup_iface_get_perf_attr_flags(wiface);
+            if (!(flags & UCT_PERF_ATTR_FLAGS_TX_RX_SHARED)) {
+                continue;
+            }
 
-        am_lane = lane_desc - select_ctx->lane_descs;
+            am_lane = lane_desc - select_ctx->lane_descs;
 
-        /* Map AM lane to specific path_index */
-        ucs_assertv((lane_desc->path_index == 0) ||
-                            (lane_desc->path_index ==
-                             UCP_WIREUP_PATH_INDEX_UNDEFINED),
-                    "rsc_index=%u am_lane=%u path_index=%u",
-                    lane_desc->rsc_index, am_lane, lane_desc->path_index);
+            /* Map AM lane to specific path_index */
+            ucs_assertv((lane_desc->path_index == 0) ||
+                                (lane_desc->path_index ==
+                                 UCP_WIREUP_PATH_INDEX_UNDEFINED),
+                        "rsc_index=%u am_lane=%u path_index=%u",
+                        lane_desc->rsc_index, am_lane, lane_desc->path_index);
 
-        lane_desc->path_index = 0;
-        num_paths_override    = 1;
-        break;
+            lane_desc->path_index = 0;
+            num_paths_override    = 1;
```

Comment:
rename "num_paths_override" to "extra_path" or similar

Reply:
fixed